lib/src/functors.ml

```diff
@@ -309,6 +309,7 @@ module Make_native (B :
   type character = char
   type t = B.t
 
+  let max_string_length = Some Sys.max_string_length
```

Comment:
It does, but that's the intent. I want to get away from exposing `Functors` (although, I was struggling with the build system to figure out how). The `bigarray` version would use `Of_mutable` instead